aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -647,87 +942,125 @@ def export_tree(
 
-        progress_bar = get_progress_bar(total=len(all_node_uuids), disable=silent)
-        pbar_base_str = 'Exporting repository - '
+    :param traversal_rules: graph traversal rules. See :const:`aiida.common.links.GraphTraversalRules` what rule names
+        are toggleable and what the defaults are.
 
-        for uuid in all_node_uuids:
-            sharded_uuid = export_shard_uuid(uuid)
+    :raises `~aiida.tools.importexport.common.exceptions.ArchiveExportError`: if there are any internal errors when
+        exporting.
+    :raises `~aiida.common.exceptions.LicensingException`: if any node is licensed under forbidden license.
+    """
+    # pylint: disable=unused-argument
+    warnings.warn('function is deprecated, use `export` instead', AiidaDeprecationWarning)  # pylint: disable=no-member
+    raise NotImplementedError
 
-            progress_bar.set_description_str(f"{pbar_base_str}UUID={uuid.split('-')[0]}", refresh=False)
-            progress_bar.update()
 
-            # Important to set create=False, otherwise creates twice a subfolder. Maybe this is a bug of insert_path?
-            thisnodefolder = nodesubfolder.get_subfolder(sharded_uuid, create=False, reset_limit=True)
+def export_zip(
+    entities: Optional[Iterable[Any]] = None,
+    filename: Optional[str] = None,
+    use_compression: bool = True,
+    **kwargs: Any,
+) -> Tuple[float, ...]:
+    """Export in a zipped folder
 
-            # Make sure the node's repository folder was not deleted
-            src = RepositoryFolder(section=Repository._section_name, uuid=uuid)  # pylint: disable=protected-access
-            if not src.exists():
-                raise exceptions.ArchiveExportError(
-                    f'Unable to find the repository folder for Node with UUID={uuid} in the local repository'
-                )
+    .. deprecated:: 1.2.1
+        Support for the parameters `what` and `outfile` will be removed in `v2.0.0`.
+        Please use `entities` and `filename` instead, respectively.
 
-            # In this way, I copy the content of the folder, and not the folder itself
-            thisnodefolder.insert_path(src=src.abspath, dest_name='.')
+    :param entities: a list of entity instances; they can belong to different models/entities.
 
-    close_progress_bar(leave=False)
+    :param filename: the filename
+        (possibly including the absolute path) of the file on which to export.
 
-    # Reset logging level
-    if silent:
-        logging.disable(level=logging.NOTSET)
+    :param use_compression: Whether or not to compress the zip file.
+
+    """
+    warnings.warn('function is deprecated, use `export` instead', AiidaDeprecationWarning)  # pylint: disable=no-member
+    output_data = export(
+        entities=entities,
+        filename=filename,
+        file_format=ExportFileFormat.ZIP,
+        use_compression=use_compression,
+        **kwargs,
+    )
+    if 'time_write_stop' in output_data:
```

Comment:
just wondering whether pushing this logic of determining what happened here is the best place.
anyhow, up to you whether it is worth fixing

Reply:
well this a deprecated function anyway, this was just the easiest way to maintain the same return signature. So no I'm not going to change unless you make me lol